### DISCUSSION TITLE
In Kubernetes, Keycloak Object Doesn't Work Properly with automated AutoScaler

### DISCUSSION THREAD

**Discussion by ML-std** [keycloak/keycloak#22139] [Ideas]
I have ArgoCD installed in my k8s cluster. ArgoCD watches my repo, and AutoSyncs the data. however, sometimes it gives some problems when working with Keycloak Object. By default, keycloak Object needs to have 'instance' field determined to be autoscalable. When I give value to 'instance' field, it works fine without an automation tool like argoCD. But it needs to work like a deployment. By default, deployment can be autoscaled without 'replicas' initialized. So, I can remove from my git repo, and autoscaler makes all the jobs.

Work needs to be done:

- instance field should be optional, like deployment to hpa to autoscale.

**Comment by shawkins** [comment]
Converted this to #22151
